compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -193,8 +219,9 @@ generateSrcFromLf env = noLoc mod
     classMethodNames = Set.fromList
         [ methodName
         | LF.DefTypeSyn{..} <- NM.toList . LF.moduleSynonyms $ envMod env
-        , LF.TStruct fields <- [synType]
-        , (fieldName, _) <- fields
+        , (fieldName, _) <- case synType of
+                LF.TStruct fields -> fields
+                _ -> []
```

Comment:
🤔 it's the same, isn't it?

Reply:
Good point, I initially thought I need to match on `TUnit` here and forgot to undo the change.